src/xvectorbin/nnet3-xvector-signal-perturb-egs.cc

```diff
@@ -129,19 +130,26 @@ int main(int argc, char *argv[]) {
       SequentialNnetExampleReader noise_reader(perturb_opts.noise_egs);
       const NnetExample &noise_egs = noise_reader.Value();
       const NnetIo &noise_io = noise_egs.io[0];
-      noise_io.features.CopyToMat(noise_mat);
-       
+      noise_io.features.CopyToMat(noise_mat);       
     }
-
+ 
     for (; !example_reader.Done(); example_reader.Next(), num_read++) {
       std::string key = example_reader.Key();
       const NnetExample &input_eg = example_reader.Value();
       const NnetIo &input_eg_io = input_eg.io[0];
       NnetExample *perturb_eg = new NnetExample();
       Matrix<BaseFloat> perturb_eg_mat, 
         input_eg_mat;
-      input_eg_io.features.CopyToMat(&input_eg_mat);
-      ApplyPerturbation(perturb_opts, input_eg_mat, noise_mat, &perturb_eg_mat);
+      input_eg_io.features.CopyToMat(&input_eg_mat);      
+      
+      // add
+      if (!perturb_opts.add_noise.empty()) {
+        // We call the PerturbExample to implement adding distortion.
+        PerturbExample(perturb_opts, input_eg_mat, &perturb_eg_mat);
```

Comment:
You should not call PertubExample here, it shoule be called in nnet3-perturb-egs!